avl_tree/AVLTree.java

```diff
@@ -0,0 +1,242 @@
+    }
+
+    public boolean isEmpty() {
+       
+        return root == null;
+    }
+    
+    public int getBalance(AVLNode root) {
+       
+        if (root == null)
+            return 0;
+        return height(root.left) - height(root.right); 
+    }
+
+    public void insert(int data) {
+        
+        root = insert(root, data);
+    }
+    
+    private AVLNode rightRotate(AVLNode root) {
+        
+        AVLNode y = root.left;  
+        AVLNode temp = y.right;
+        y.right = root;
+        root.left = temp;
+        root.height = 1 + Math.max(height(root.left), height(root.right));    
+        y.height = 1 + Math.max(height(y.left), height(y.right));   
+        return y;
+    } 
+      
+    private AVLNode leftRotate(AVLNode root) {
+        
+        AVLNode y = root.right;  
+        AVLNode temp = y.left;
+        y.left = root;
+        root.right = temp;
+        root.height = 1 + Math.max(height(root.left), height(root.right));    
+        y.height = 1 + Math.max(height(y.left), height(y.right));   
+        return y;
+    } 
+    
+    private AVLNode insert(AVLNode root, int data) {
+        
+        if (root == null) {
+            root = new AVLNode(data);
+        }
+        else {   
+              if (data < root.data) { 
+              root.left = insert(root.left, data);
+             }
+              else if (data > root.data) {
+                root.right = insert(root.right, data);
+            } 
+              else {
+                    return root;
+              }        
+        }
+        root.height = 1 + Math.max(height(root.right), height(root.left));  
+        int balance = height(root.left) - height(root.right); 
+            
+        if (balance > 1 && getBalance(root.left) >= 0)  //left-left case
+            return rightRotate(root);
+        
+        else if (balance < -1 && getBalance(root.right) < 0)  //right-right case 
+            return leftRotate(root);
+        
+        else if (balance > 1 && getBalance(root.right) < 0) {   //left-right case
+            root.left = leftRotate(root.left);   
+            return rightRotate(root);
+        }
+        
+        else if (balance < -1 && getBalance(root.left) >= 0) {  //right-left case 
+            root.right = rightRotate(root.right);
+            return leftRotate(root);   
+        } 
+        return root;
+    }
+    
+    public boolean search(int val) { 
+   
+        return search(root, val);
+    }
+    
+    private boolean search(AVLNode root, int val) {
+    
+        boolean found = false;
+        while ((root != null) && !found) {
+            int rval = root.data;
+            if (val < rval)
+                root = root.left;
+            else if (val > rval)
+                root = root.right;
+            else {
+                found = true;
+                break;
+            }
+            found = search(root, val);
+        }
+        return found;
+    }
+     
+    public void inorder() {
```

Comment:
I think the method name here should be `inorderPrint()` as you are trying to print the AVL tree. your views?